contrail_api_cli/commands/batch.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import shlex
+import sys
+import fileinput
+from six import text_type
+
+from keystoneauth1.exceptions.http import HttpError, HTTPClientError
+
+from ..command import Command, Arg, Option
+from ..manager import CommandManager
+from ..context import Context
+from ..schema import SchemaError
+from ..exceptions import CommandError, CommandNotFound, NotFound, Exists
+from ..utils import printo
+
+
+class Batch(Command):
+    """ Execute commands from file(s) or from stdin
+    """
+    description = "Run commands from a batch file/stdin"
+    files = Arg(nargs="*", help="Name of the batch file")
+
+    def __call__(self, files=None):
+        manager = CommandManager()
+        try:
+            for line in fileinput.input(files=files):
+                if line[0] == '#':
+                    continue
+                action = shlex.split(line.rstrip())
+                if len(action) < 1:
+                    continue
+                cmd = manager.get(action[0])
+                args = action[1:]
+                result = cmd.parse_and_call(*args)
+                if result:
+                    printo(result)
+        except IOError:
+            printo("Cannot read from file: {}".format(filename))
+        except (HttpError, HTTPClientError, CommandError, CommandNotFound,
+                SchemaError, NotFound, Exists) as e:
+            printo(text_type(e))
```

Comment:
Same here

Reply:
Actually you don't need to catch anything here, theses exceptions are already catched in main.py